pkg/frontend/querymiddleware/spin_off_subqueries.go

```diff
@@ -95,19 +91,27 @@ func newSpinOffSubqueriesMiddleware(
 	limits Limits,
 	logger log.Logger,
 	engine *promql.Engine,
-	rangeHandler MetricsQueryHandler,
 	registerer prometheus.Registerer,
+	rangeMiddlewares []MetricsQueryMiddleware,
```

Comment:
let's take a single `MetricsQueryMiddleware` and use `middleware.Merge` (but again, i'm not sure we even need to merge anything if we want to add only the split-and-cache middleware)